volatility3/framework/plugins/linux/pagecache.py

```diff
@@ -186,8 +186,12 @@ def _walk_dentry(
 
             seen_dentries.add(dentry_addr)
 
-            inode = dentry.d_inode
-            if not (inode and inode.is_valid()):
+            inode_ptr = dentry.d_inode
+            if not (inode_ptr and inode_ptr.is_readable()):
+                continue
+
+            inode = inode_ptr.dereference()
```

Comment:
Is this not already covered by `inode_ptr.is_readable`?  How do the two differ?

Reply:
hm not sure which line are you referring to, I see this comment on line 193 which is dereferencing the pointer. This will raise an exception on invalid pointers for which you need the previous checks.
- if `inode_ptr`: only checks for `inode_ptr == 0` or `None` as we discussed in https://github.com/volatilityfoundation/volatility3/pull/1243.
- inode_ptr.is_readable(): checks that the pointer is valid in its address space.
- inode.is_valid(): Checks that the content of the inode is valid.

All of them are needed.

Reply:
Ok, just figured `is_readable` would be enough, but if they're all needed that's fine.